### ISSUE TITLE
[bitnami/redis] : Redis not getting deployed on provided NodePort rather taking random ports

### ISSUE THREAD

**Issue by pritpambhar** [bitnami/charts#30082]
### Name and Version

bitnami/redis-20.1.5

### What architecture are you using?

None

### What steps will reproduce the bug?

Firstly I am deploying redis with clusterIP but In development env, I used to patch redis service with kubectl command mentioned below like this.
```
'kubectl patch svc redis -p '{"spec":{"ports:[{"nodePort":30100,"port":6379,"targetPort":6379,"name":"redis","protocol":"TCP"}],"type":"NodePort"}}' -n xxxnamespace'.
```
So through this I am able to patch the redis svc and I can see service something like this.
```
redis           NodePort    10.110.xx.xxx    <none>        6379:30100/TCP, 26379:31913/TCP   7d17h
```
Now,I wanted to achieve same with config so when I deploy redis with below values,
```
sentinel.service.type = NodePort
sentinel.service.nodePorts.redis="30100"
```
I am getting services details as below
```
redis           NodePort    10.110.xx.xxx    <none>        30100:30100/TCP,30426:30426/TCP,26379:32138/TCP,6379:32569/TCP   7d17h
redis-node-0    NodePort    10.103.xx.xxx    <none>        30427:30427/TCP,30101:30101/TCP,26379:31649/TCP,6379:31682/TCP   73s
redis-node-1    NodePort    10.98.xx.xxx    <none>        30428:30428/TCP,30102:30102/TCP,26379:32751/TCP,6379:30358/TCP   73s
redis-node-2    NodePort    10.103.xx.xxx    <none>        30429:30429/TCP,30103:30103/TCP,26379:32668/TCP,6379:31865/TCP   73s
```
So basically, it is not mapping redis svc ports 30100 to 6379 as expected and also creating multiple services and other ports also.
I want to achieve similar to what I am doing with kubectl command.

### What is the expected behavior?

redis           NodePort    10.110.xx.xxx    <none>        6379:30100/TCP, 26379:31913/TCP   7d17h

### What do you see instead?
```
redis           NodePort    10.110.xx.xxx    <none>        30100:30100/TCP,30426:30426/TCP,26379:32138/TCP,6379:32569/TCP   7d17h
redis-node-0    NodePort    10.103.xx.xxx    <none>        30427:30427/TCP,30101:30101/TCP,26379:31649/TCP,6379:31682/TCP   73s
redis-node-1    NodePort    10.98.xx.xxx    <none>        30428:30428/TCP,30102:30102/TCP,26379:32751/TCP,6379:30358/TCP   73s
redis-node-2    NodePort    10.103.xx.xxx    <none>        30429:30429/TCP,30103:30103/TCP,26379:32668/TCP,6379:31865/TCP   73s
```

**Comment by javsalgar** [comment]
Hi,

Apart from changing 

```
sentinel.service.type = NodePort
sentinel.service.nodePorts.redis="30100"
``` 

Did you change any other values from the default?

**Comment by pritpambhar** [comment]
Below is the redis-value.yaml file I am using and I am also setting some value through helmfile which are mentioned after this code snippet.

redis-value.yaml
```
global:
  imageRegistry: ""
  imagePullSecrets: []
  defaultStorageClass: ""
  storageClass: ""
  redis:
    password: ""
  compatibility:
    openshift:
      adaptSecurityContext: auto
kubeVersion: ""
nameOverride: ""
fullnameOverride: ""
namespaceOverride: ""
commonLabels: {}
commonAnnotations: {}
secretAnnotations: {}
clusterDomain: cluster.local
extraDeploy: []
useHostnames: false
nameResolutionThreshold: 5
nameResolutionTimeout: 5
diagnosticMode:
  enabled: false
  command:
    - sleep
  args:
    - infinity
image:
  registry: --IMAGE_REGISTRY--
  repository: prod-ext-redis
  tag: 7.4.0-debian-12-r4
  digest: ""
  pullPolicy: IfNotPresent
  pullSecrets: []
  debug: false
architecture: replication
auth:
  enabled: true
  sentinel: true
  password: ""
  existingSecret: redis-secret
  existingSecretPasswordKey: REDIS_PASSWORD
  usePasswordFiles: false
  usePasswordFileFromSecret: false
commonConfiguration: |-
  appendonly yes
  save ""
existingConfigmap: ""
master:
  count: 1
  revisionHistoryLimit: 10
  configuration: ""
  disableCommands:
    - FLUSHDB
    - FLUSHALL
  command: []
  args: []
  enableServiceLinks: true
  preExecCmds: []
  extraFlags: []
  extraEnvVars: []
  extraEnvVarsCM: ""
  extraEnvVarsSecret: ""
  containerPorts:
    redis: 6379
  startupProbe:
    enabled: false
    initialDelaySeconds: 20
    periodSeconds: 5
    timeoutSeconds: 5
    successThreshold: 1
    failureThreshold: 5
  livenessProbe:
    enabled: true
    initialDelaySeconds: 20
    periodSeconds: 5
    timeoutSeconds: 5
    successThreshold: 1
    failureThreshold: 5
  readinessProbe:
    enabled: true
    initialDelaySeconds: 20
    periodSeconds: 5
    timeoutSeconds: 1
    successThreshold: 1
    failureThreshold: 5
  customStartupProbe: {}
  customLivenessProbe: {}
  customReadinessProbe: {}
  resourcesPreset: "nano"
  resources:
    limits:
       cpu: 250m
       memory: 256Mi
    requests:
       cpu: 250m
       memory: 256Mi
  podSecurityContext:
    enabled: false
    fsGroupChangePolicy: Always
    sysctls: []
    supplementalGroups: []
    fsGroup: 1001
  containerSecurityContext:
    enabled: true
    runAsNonRoot: true
  kind: StatefulSet
  schedulerName: ""
  updateStrategy:
    type: RollingUpdate
  minReadySeconds: 0
  priorityClassName: ""
  automountServiceAccountToken: false
  hostAliases: []
  podLabels: {}
  podAnnotations: {}
  shareProcessNamespace: false
  podAffinityPreset: ""
  podAntiAffinityPreset: soft
  nodeAffinityPreset:
    type: ""
    key: ""
    values: []
  affinity: {}
  nodeSelector: {}
  tolerations: []
  topologySpreadConstraints: []
  dnsPolicy: ""
  dnsConfig: {}
  lifecycleHooks: {}
  extraVolumes: []
  extraVolumeMounts: []
  sidecars: []
  initContainers: []
  persistence:
    enabled: false
    medium: ""
    sizeLimit: ""
    path: /data
    subPath: ""
    subPathExpr: ""
    storageClass: ""
    accessModes:
      - ReadWriteOnce
    size: 8Gi
    annotations: {}
    labels: {}
    selector: {}
    dataSource: {}
    existingClaim: ""
  persistentVolumeClaimRetentionPolicy:
    enabled: false
    whenScaled: Retain
    whenDeleted: Retain
  service:
    type: ClusterIP
    portNames:
      redis: "tcp-redis"
    ports:
      redis: 6379
    nodePorts:
      redis: ""
    externalTrafficPolicy: Cluster
    extraPorts: []
    internalTrafficPolicy: Cluster
    clusterIP: ""
    loadBalancerIP: ""
    loadBalancerClass: ""
    loadBalancerSourceRanges: []
    externalIPs: []
    annotations: {}
    sessionAffinity: None
    sessionAffinityConfig: {}
  terminationGracePeriodSeconds: 30
  serviceAccount:
    create: false
    name: ""
    automountServiceAccountToken: true
    annotations: {}
  pdb:
    create: false
    minAvailable: 1
    maxUnavailable: ""
replica:
  kind: StatefulSet
  replicaCount: 3
  revisionHistoryLimit: 10
  configuration: ""
  disableCommands:
    - FLUSHDB
    - FLUSHALL
  command: []
  args: []
  enableServiceLinks: true
  preExecCmds: []
  extraFlags: []
  extraEnvVars: []
  extraEnvVarsCM: ""
  extraEnvVarsSecret: ""
  externalMaster:
    enabled: false
    host: ""
    port: 6379
  containerPorts:
    redis: 6379
  startupProbe:
    enabled: true
    initialDelaySeconds: 5
    periodSeconds: 10
    timeoutSeconds: 5
    successThreshold: 1
    failureThreshold: 22
  livenessProbe:
    enabled: true
    initialDelaySeconds: 5
    periodSeconds: 5
    timeoutSeconds: 5
    successThreshold: 1
    failureThreshold: 5
  readinessProbe:
    enabled: true
    initialDelaySeconds: 5
    periodSeconds: 5
    timeoutSeconds: 1
    successThreshold: 1
    failureThreshold: 5
  customStartupProbe: {}
  customLivenessProbe: {}
  customReadinessProbe: {}
  resourcesPreset: "nano"
  resources:
    limits:
       cpu: 250m
       memory: 256Mi
    requests:
       cpu: 250m
       memory: 256Mi
  podSecurityContext:
    enabled: false
    fsGroupChangePolicy: Always
    sysctls: []
    supplementalGroups: []
    fsGroup: 1001
  containerSecurityContext:
    enabled: true
    runAsNonRoot: true
  schedulerName: ""
  updateStrategy:
    type: RollingUpdate
  minReadySeconds: 0
  priorityClassName: ""
  podManagementPolicy: ""
  automountServiceAccountToken: false
  hostAliases: []
  podLabels: {}
  podAnnotations: {}
  shareProcessNamespace: false
  podAffinityPreset: ""
  podAntiAffinityPreset: soft
  nodeAffinityPreset:
    type: ""
    key: ""
    values: []
  affinity: {}
  nodeSelector: {}
  tolerations: []
  topologySpreadConstraints: []
  dnsPolicy: ""
  dnsConfig: {}
  lifecycleHooks: {}
  extraVolumes: []
  extraVolumeMounts: []
  sidecars: []
  initContainers: []
  persistence:
    enabled: false
    medium: ""
    sizeLimit: ""
    path: /data
    subPath: ""
    subPathExpr: ""
    storageClass: ""
    accessModes:
      - ReadWriteOnce
    size: 8Gi
    annotations: {}
    labels: {}
    selector: {}
    dataSource: {}
    existingClaim: ""
  persistentVolumeClaimRetentionPolicy:
    enabled: false
    whenScaled: Retain
    whenDeleted: Retain
  service:
    type: ClusterIP
    ports:
      redis: 6379
    nodePorts:
      redis: ""
    externalTrafficPolicy: Cluster
    internalTrafficPolicy: Cluster
    extraPorts: []
    clusterIP: ""
    loadBalancerIP: ""
    loadBalancerClass: ""
    loadBalancerSourceRanges: []
    annotations: {}
    sessionAffinity: None
    sessionAffinityConfig: {}
  terminationGracePeriodSeconds: 30
  autoscaling:
    enabled: false
    minReplicas: 1
    maxReplicas: 11
    targetCPU: ""
    targetMemory: ""
  serviceAccount:
    create: false
    name: ""
    automountServiceAccountToken: true
    annotations: {}
  pdb:
    create: false
    minAvailable: 1
    maxUnavailable: ""
sentinel:
  enabled: true
  image:
    registry: --IMAGE_REGISTRY--
    repository: prod-ext-redis-sentinel
    tag: 7.4.0-debian-12-r5
    digest: ""
    pullPolicy: IfNotPresent
    pullSecrets: []
    debug: false
  annotations: {}
  masterSet: mymaster
  quorum: 2
  getMasterTimeout: 40
  automateClusterRecovery: false
  redisShutdownWaitFailover: true
  downAfterMilliseconds: 20000
  failoverTimeout: 18000
  parallelSyncs: 1
  configuration: ""
  command: []
  args: []
  enableServiceLinks: true
  preExecCmds: []
  extraEnvVars: []
  extraEnvVarsCM: ""
  extraEnvVarsSecret: ""
  externalMaster:
    enabled: false
    host: ""
    port: 6379
  containerPorts:
    sentinel: 26379
  startupProbe:
    enabled: true
    initialDelaySeconds: 5
    periodSeconds: 10
    timeoutSeconds: 5
    successThreshold: 1
    failureThreshold: 22
  livenessProbe:
    enabled: true
    initialDelaySeconds: 5
    periodSeconds: 5
    timeoutSeconds: 5
    successThreshold: 1
    failureThreshold: 5
  readinessProbe:
    enabled: true
    initialDelaySeconds: 5
    periodSeconds: 5
    timeoutSeconds: 1
    successThreshold: 1
    failureThreshold: 5
  customStartupProbe: {}
  customLivenessProbe: {}
  customReadinessProbe: {}
  persistence:
    enabled: false
    storageClass: ""
    accessModes:
      - ReadWriteOnce
    size: 100Mi
    annotations: {}
    labels: {}
    selector: {}
    dataSource: {}
    medium: ""
    sizeLimit: ""
  persistentVolumeClaimRetentionPolicy:
    enabled: false
    whenScaled: Retain
    whenDeleted: Retain
  resourcesPreset: "nano"
  resources:
    limits:
       cpu: 200m
       memory: 128Mi
    requests:
       cpu: 200m
       memory: 128Mi
  containerSecurityContext:
    enabled: true
    runAsNonRoot: true
  lifecycleHooks: {}
  extraVolumes: []
  extraVolumeMounts: []
  service:
    type: NodePort
    ports:
      redis: 6379
      sentinel: 26379
    nodePorts:
      redis: "30100"
      sentinel: "30426"
    externalTrafficPolicy: Cluster
    extraPorts: []
    clusterIP: ""
    createMaster: false
    loadBalancerIP: ""
    loadBalancerClass: ""
    loadBalancerSourceRanges: []
    annotations: {}
    sessionAffinity: None
    sessionAffinityConfig: {}
    headless:
      annotations: {}
  masterService:
    enabled: false
    type: ClusterIP
    ports:
      redis: 6379
    nodePorts:
      redis: ""
    externalTrafficPolicy: ""
    extraPorts: []
    clusterIP: ""
    loadBalancerIP: ""
    loadBalancerClass: ""
    loadBalancerSourceRanges: []
    annotations: {}
    sessionAffinity: None
    sessionAffinityConfig: {}
  terminationGracePeriodSeconds: 30
serviceBindings:
  enabled: false
networkPolicy:
  enabled: false
  allowExternal: true
  allowExternalEgress: true
  extraIngress: []
  extraEgress: []
  ingressNSMatchLabels: {}
  ingressNSPodMatchLabels: {}
  metrics:
    allowExternal: true
    ingressNSMatchLabels: {}
    ingressNSPodMatchLabels: {}
podSecurityPolicy:
  create: false
  enabled: false
rbac:
  create: false
  rules: []
serviceAccount:
  create: false
  name: ""
  automountServiceAccountToken: true
  annotations: {}
pdb: {}
tls:
  enabled: false
  authClients: true
  autoGenerated: false
  existingSecret: ""
  certificatesSecret: ""
  certFilename: ""
  certKeyFilename: ""
  certCAFilename: ""
  dhParamsFilename: ""
metrics:
  enabled: false
  image:
    registry: docker.io
    repository: bitnami/redis-exporter
    tag: 1.61.0-debian-12-r2
    digest: ""
    pullPolicy: IfNotPresent
    pullSecrets: []
  containerPorts:
    http: 9121
  startupProbe:
    enabled: false
    initialDelaySeconds: 10
    periodSeconds: 10
    timeoutSeconds: 5
    successThreshold: 1
    failureThreshold: 5
  livenessProbe:
    enabled: false
    initialDelaySeconds: 10
    periodSeconds: 10
    timeoutSeconds: 5
    successThreshold: 1
    failureThreshold: 5
  readinessProbe:
    enabled: false
    initialDelaySeconds: 5
    periodSeconds: 10
    timeoutSeconds: 1
    successThreshold: 1
    failureThreshold: 3
  customStartupProbe: {}
  customLivenessProbe: {}
  customReadinessProbe: {}
  command: []
  redisTargetHost: "localhost"
  extraArgs: {}
  extraEnvVars: []
  containerSecurityContext:
    enabled: true
    seLinuxOptions: {}
    runAsUser: 1001
    runAsGroup: 1001
    runAsNonRoot: true
    allowPrivilegeEscalation: false
    readOnlyRootFilesystem: true
    seccompProfile:
      type: RuntimeDefault
    capabilities:
      drop: ["ALL"]
  extraVolumes: []
  extraVolumeMounts: []
  resourcesPreset: "nano"
  resources: {}
  podLabels: {}
  podAnnotations:
    prometheus.io/scrape: "true"
    prometheus.io/port: "9121"
  service:
    enabled: true
    type: ClusterIP
    ports:
      http: 9121
    externalTrafficPolicy: Cluster
    extraPorts: []
    loadBalancerIP: ""
    loadBalancerClass: ""
    loadBalancerSourceRanges: []
    annotations: {}
    clusterIP: ""
  serviceMonitor:
    port: http-metrics
    enabled: false
    namespace: ""
    interval: 30s
    scrapeTimeout: ""
    relabelings: []
    relabellings: []
    metricRelabelings: []
    honorLabels: false
    additionalLabels: {}
    podTargetLabels: []
    sampleLimit: false
    targetLimit: false
    additionalEndpoints: []
  podMonitor:
    port: metrics
    enabled: false
    namespace: ""
    interval: 30s
    scrapeTimeout: ""
    relabelings: []
    relabellings: []
    metricRelabelings: []
    honorLabels: false
    additionalLabels: {}
    podTargetLabels: []
    sampleLimit: false
    targetLimit: false
    additionalEndpoints: []
  prometheusRule:
    enabled: false
    namespace: ""
    additionalLabels: {}
    rules: []
volumePermissions:
  enabled: false
  image:
    registry: docker.io
    repository: bitnami/os-shell
    tag: 12-debian-12-r27
    digest: ""
    pullPolicy: IfNotPresent
    pullSecrets: []
  resourcesPreset: "nano"
  resources: {}
  containerSecurityContext:
    seLinuxOptions: {}
    runAsUser: 0
  extraEnvVars: []
kubectl:
  image:
    registry: --IMAGE_REGISTRY--
    repository: prod-ext-kubectl
    tag: 1.31.1-debian-12-r2
    digest: ""
    pullPolicy: IfNotPresent
    pullSecrets: []
  command: ["/opt/bitnami/scripts/kubectl-scripts/update-master-label.sh"]
  containerSecurityContext:
    enabled: true
    runAsNonRoot: true
  resources:
    limits: {}
    requests: {}
sysctl:
  enabled: false
  image:
    registry: docker.io
    repository: bitnami/os-shell
    tag: 12-debian-12-r27
    digest: ""
    pullPolicy: IfNotPresent
    pullSecrets: []
  command: []
  mountHostSys: false
  resourcesPreset: "nano"
  resources: {}
useExternalDNS:
  enabled: false
  suffix: ""
  annotationKey: external-dns.alpha.kubernetes.io/
  additionalAnnotations: {}
```

helmfile.yaml

```
set:
  - name: serviceAccount.create
    value: false
  - name: serviceAccount.name
    value: "solumina-external"
  - name : image.registry 
    value: {{ $.Values.global.image.registry }}
  - name: image.repository
    value: prod-ext-redis-stack-server
  - name: image.tag
    value: 7.4.0-rc2
  - name : image.pullPolicy 
    value: {{ $.Values.global.image.pullPolicy }}     
  - name : sentinel.image.registry 
    value: {{ $.Values.global.image.registry }}
  - name: sentinel.image.repository
    value: prod-ext-redis-sentinel
  - name: sentinel.image.tag
    value: 7.4.0-debian-12-r5
  - name : sentinel.image.pullPolicy 
    value: {{ $.Values.global.image.pullPolicy }}
  - name: kubectl.image.registry
    value: {{ $.Values.global.image.registry }}
  - name: kubectl.image.repository
    value: prod-ext-kubectl
  - name: kubectl.image.tag
    value: 1.31.1-debian-12-r2
  - name: kubectl.image.pullPolicy 
    value: {{ $.Values.global.image.pullPolicy }}
  - name: master.extraFlags
    values: 
    - "--loadmodule /opt/redis-stack/lib/rejson.so"
    - "--loadmodule /opt/redis-stack/lib/redisearch.so"
  - name: replica.extraFlags
    values: 
    - "--loadmodule /opt/redis-stack/lib/rejson.so"
    - "--loadmodule /opt/redis-stack/lib/redisearch.so"
  - name: master.containerSecurityContext.runAsUser
    value: {{ .Values.global.FILESHARE_UID }}
  - name: master.containerSecurityContext.runAsGroup
    value: {{ .Values.global.FILESHARE_GID }}
  - name: replica.containerSecurityContext.runAsUser
    value: {{ .Values.global.FILESHARE_UID }}
  - name: replica.containerSecurityContext.runAsGroup
    value: {{ .Values.global.FILESHARE_GID }}
  - name: sentinel.containerSecurityContext.runAsUser
    value: {{ .Values.global.FILESHARE_UID }}
  - name: sentinel.containerSecurityContext.runAsGroup
    value: {{ .Values.global.FILESHARE_GID }}
  - name: kubectl.containerSecurityContext.runAsUser
    value: {{ .Values.global.FILESHARE_UID }}
  - name: kubectl.containerSecurityContext.runAsGroup
    value: {{ .Values.global.FILESHARE_GID }}
```

**Comment by juan131** [comment]
Hi @pritpambhar 

It's not very clear to me whether you want to modify the node port for Redis Sentinel service or the Redis replica service. In any case, I give it a try using the `values.yaml` below:

```yaml
architecture: "replication"
sentinel:
  enabled: true
  service:
    type: "NodePort"
    nodePorts:
      redis: "30100"
```

And.. i found no issues with the node port to be used:

```console
$ helm template redis oci://registry-1.docker.io/bitnamicharts/redis -f values.yaml -s templates/sentinel/service.yaml
(...)
# Source: redis/templates/sentinel/service.yaml
apiVersion: v1
kind: Service
metadata:
  name: redis
  namespace: "test"
  labels:
    app.kubernetes.io/instance: redis
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: redis
    app.kubernetes.io/version: 7.4.1
    helm.sh/chart: redis-20.2.1
    app.kubernetes.io/component: node
spec:
  type: NodePort
  externalTrafficPolicy: "Cluster"
  sessionAffinity: None
  ports:
    - name: tcp-redis
      port: 30100
      targetPort: 6379
      nodePort: 30100
    - name: tcp-sentinel
      port:  0
      targetPort: 26379
      nodePort:  0
    - name: sentinel-internal
      nodePort: null
      port: 26379
      protocol: TCP
      targetPort: 26379
    - name: redis-internal
      nodePort: null
      port: 6379
      protocol: TCP
      targetPort: 6379
  selector:
    app.kubernetes.io/instance: redis
    app.kubernetes.io/name: redis
    app.kubernetes.io/component: node
```

**Comment by pritpambhar** [comment]
Hi @juan131,
So basically when I am deploying redis service as a clusterIP, I am able to see only one service as below. I am assuming its a master service.

redis    ClusterIP    10.119.xx.xxx    <none>    6376/TCP, 26379/TCP    12d


Now I want to deploy this service as a NodePort on 30100. So when I am changing the master.service.type and master.service.nodePorts.redis values it is not changing anything to this particular redis service. But when I am changing sentinel.service.type to "NodePort", this Atleast changes my redis service to NodePort but not on desired nodeport and also it is  creating 3 other services as mentioned in description.
Basically I want above service to be get deployed on NodePort with 30100 mapped to 6379 (redis) and 26379(sentinel) to any random port.

**Comment by juan131** [comment]
Hi @pritpambhar 

I'm sorry but I'm a little bit confused about your setup, could you please provide the chart parameters (provided via `values.yaml` or using the `--set` flag) you used to install your Helm release? Please exclusively provide the parameters you customized avoiding the ones with default values, otherwise the output is very noisy and it's hard for us to identify the issue.

> Note: you can easily obtain the above parameters using `helm get values RELEASE_NAME`

**Comment by juan131** [comment]
Hi @pritpambhar 

Here's a reduced version with what I think you actually changed, please let me know if I'm right:

```yaml
auth:
  enabled: true
  sentinel: true
  existingSecret: redis-secret
  existingSecretPasswordKey: REDIS_PASSWORD
  password: ""
image:
  registry: --IMAGE_REGISTRY--
  repository: prod-ext-redis
  tag: 7.4.0-debian-12-r4
master:
  persistence:
    enabled: false
  containerSecurityContext:
    enabled: true
    runAsNonRoot: true
  podSecurityContext:
    enabled: false
  resources:
    limits:
      cpu: 250m
      memory: 256Mi
    requests:
      cpu: 250m
      memory: 256Mi
  service:
    nodePorts:
      redis: "30100"
    type: NodePort
  serviceAccount:
    create: false
networkPolicy:
  enabled: false
replica:
  persistence:
    enabled: false
  containerSecurityContext:
    enabled: true
    runAsNonRoot: true
  podSecurityContext:
    enabled: false
  replicaCount: 3
  resources:
    limits:
      cpu: 250m
      memory: 256Mi
    requests:
      cpu: 250m
      memory: 256Mi
  resourcesPreset: nano
  service:
    type: ClusterIP
  serviceAccount:
    create: false
sentinel:
  enabled: true
  containerSecurityContext:
    enabled: true
    runAsNonRoot: true
  image:
    registry: --IMAGE_REGISTRY--
    repository: prod-ext-redis-sentinel
    tag: 7.4.0-debian-12-r5
  masterService:
    nodePorts:
      redis: "30100"
    type: NodePort
  persistence:
    enabled: false
  resources:
    limits:
      cpu: 200m
      memory: 128Mi
    requests:
      cpu: 200m
      memory: 128Mi
  service:
    type: ClusterIP
serviceAccount:
  create: false
```

If I use your values to render the chart, I obtain the services below:

```console
$ helm template redis oci://registry-1.docker.io/bitnamicharts/redis -f redis-values.yaml -s templates/*svc* -s templates/**/*service*
(...)
# Source: redis/templates/headless-svc.yaml
apiVersion: v1
kind: Service
metadata:
  name: redis-headless
  namespace: "test"
  labels:
    app.kubernetes.io/instance: redis
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: redis
    app.kubernetes.io/version: 7.4.1
    helm.sh/chart: redis-20.2.1
spec:
  type: ClusterIP
  clusterIP: None
  publishNotReadyAddresses: true
  ports:
    - name: tcp-redis
      port: 6379
      targetPort: redis
    - name: tcp-sentinel
      port: 26379
      targetPort: redis-sentinel
  selector:
    app.kubernetes.io/instance: redis
    app.kubernetes.io/name: redis
---
# Source: redis/templates/sentinel/service.yaml
apiVersion: v1
kind: Service
metadata:
  name: redis
  namespace: "test"
  labels:
    app.kubernetes.io/instance: redis
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: redis
    app.kubernetes.io/version: 7.4.1
    helm.sh/chart: redis-20.2.1
    app.kubernetes.io/component: node
spec:
  type: ClusterIP
  sessionAffinity: None
  ports:
    - name: tcp-redis
      port: 6379
      targetPort: 6379
      nodePort: null
    - name: tcp-sentinel
      port: 26379
      targetPort: 26379
      nodePort: null
  selector:
    app.kubernetes.io/instance: redis
    app.kubernetes.io/name: redis
    app.kubernetes.io/component: node
```

We can see the `30100` value was ignored and the reason is that sentinel is enabled. However, if we apply this patch in your values:

```diff
diff --git a/redis-values.yaml b/redis-values.yaml
index 20c3c00a81..5cabe6cd3e 100644
--- a/redis-values.yaml
+++ b/redis-values.yaml
@@ -32,6 +32,7 @@ master:
 networkPolicy:
   enabled: false
 replica:
+  automountServiceAccountToken: true
   persistence:
     enabled: false
   containerSecurityContext:
@@ -62,6 +63,7 @@ sentinel:
     repository: prod-ext-redis-sentinel
     tag: 7.4.0-debian-12-r5
   masterService:
+    enabled: true
     nodePorts:
       redis: "30100"
     type: NodePort
@@ -77,4 +79,6 @@ sentinel:
   service:
     type: ClusterIP
 serviceAccount:
-  create: false
+  create: true
+rbac:
+  create: true
```

And we render again the templates:

```console
$ helm template redis oci://registry-1.docker.io/bitnamicharts/redis -f redis-values.yaml -s templates/*svc* -s templates/**/*service*
(...)
---
# Source: redis/templates/headless-svc.yaml
apiVersion: v1
kind: Service
metadata:
  name: redis-headless
  namespace: "test"
  labels:
    app.kubernetes.io/instance: redis
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: redis
    app.kubernetes.io/version: 7.4.1
    helm.sh/chart: redis-20.2.1
spec:
  type: ClusterIP
  clusterIP: None
  publishNotReadyAddresses: true
  ports:
    - name: tcp-redis
      port: 6379
      targetPort: redis
    - name: tcp-sentinel
      port: 26379
      targetPort: redis-sentinel
  selector:
    app.kubernetes.io/instance: redis
    app.kubernetes.io/name: redis
---
# Source: redis/templates/sentinel/service.yaml
apiVersion: v1
kind: Service
metadata:
  name: redis
  namespace: "test"
  labels:
    app.kubernetes.io/instance: redis
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: redis
    app.kubernetes.io/version: 7.4.1
    helm.sh/chart: redis-20.2.1
    app.kubernetes.io/component: node
spec:
  type: ClusterIP
  sessionAffinity: None
  ports:
    - name: tcp-redis
      port: 6379
      targetPort: 6379
      nodePort: null
    - name: tcp-sentinel
      port: 26379
      targetPort: 26379
      nodePort: null
  selector:
    app.kubernetes.io/instance: redis
    app.kubernetes.io/name: redis
    app.kubernetes.io/component: node
---
# Source: redis/templates/sentinel/service.yaml
apiVersion: v1
kind: Service
metadata:
  name: "redis-master"
  namespace: "test"
  labels:
    app.kubernetes.io/instance: redis
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: redis
    app.kubernetes.io/version: 7.4.1
    helm.sh/chart: redis-20.2.1
    app.kubernetes.io/component: node
spec:
  type: NodePort
  externalTrafficPolicy: ""
  sessionAffinity: None
  ports:
    - name: tcp-redis
      port: 30100
      targetPort: 6379
      nodePort: 30100
  selector:
    app.kubernetes.io/instance: redis
    app.kubernetes.io/name: redis
    isMaster: "true"
```

We can see that now there's a `redis-master` service and the nodePort used for this service is `30100`

**Comment by pritpambhar** [comment]
Hi @juan131,
After applying the suggested changes, I am getting services as something below. 
Ideal expectation is to convert 'redis' service on NodePort at 30100. Also unable to connect 'redis-master' on 30100 externally. 
```
redis                             ClusterIP   10.110.22.231    <none>        6379/TCP,26379/TCP   20d
redis-master                      NodePort    10.108.59.59     <none>        30100:30100/TCP      119m
```

**Comment by juan131** [comment]
Hi @pritpambhar 

I'm not sure what you're proposing, do you want both services to be exposed using the same node port? That's not possible

**Comment by pritpambhar** [comment]
Hi @juan131 ,
I don't want to both service, I just want 'redis' service to get exposed on 30100.

**Comment by juan131** [comment]
Then you shouldn't set `sentinel.masterService` to `true` and set the custom node port at `sentinel.service.nodePorts.redis`

**Comment by pritpambhar** [comment]
HI @juan131 ,
When I am assigning value as below in my helmfile.yaml I am getting below error

```
sentinel.service.nodePorts.redis="30100"
sentinel.service.nodePorts.sentinel="32456"
```

ERROR :-

```
- sentinel.service.nodePorts.redis: Invalid type. Expected: string, given: integer
- sentinel.service.nodePorts.sentinel: Invalid type. Expected: string, given: integer
```

**Comment by juan131** [comment]
@pritpambhar try using:

- `--set-string sentinel.service.nodePorts.redis="1234"`

instead of

- `--set sentinel.service.nodePorts.redis="1234"`

**Comment by juan131** [comment]
I'm afraid I'm not familiar with that tool @pritpambhar. I'd recommend you to contact `helmfile` developers and ask them about the equivalent way to pass a number as string using that tool (equivalent to `--set-string` flag in `helm`)

**Comment by pritpambhar** [comment]
For someone facing the similar issue, they can refer the below link.
[https://github.com/helmfile/helmfile/issues/1801]
